internal/services/firewall/firewall_application_rule_collection_resource.go

```diff
@@ -326,34 +324,30 @@ func resourceFirewallApplicationRuleCollectionDelete(d *pluginsdk.ResourceData,
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := azure.ParseAzureResourceID(d.Id())
+	id, err := parse.FirewallApplicationRuleCollectionID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resourceGroup := id.ResourceGroup
-	firewallName := id.Path["azureFirewalls"]
-	name := id.Path["applicationRuleCollections"]
-
-	locks.ByName(firewallName, azureFirewallResourceName)
-	defer locks.UnlockByName(firewallName, azureFirewallResourceName)
+	locks.ByName(id.AzureFirewallName, azureFirewallResourceName)
+	defer locks.UnlockByName(id.AzureFirewallName, azureFirewallResourceName)
 
-	firewall, err := client.Get(ctx, resourceGroup, firewallName)
+	firewall, err := client.Get(ctx, id.ResourceGroup, id.AzureFirewallName)
 	if err != nil {
 		if utils.ResponseWasNotFound(firewall.Response) {
 			// assume deleted
 			return nil
 		}
 
-		return fmt.Errorf("making Read request on Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+		return fmt.Errorf("making Read request on Azure Firewall %q : %+v", id, err)
```

Comment:
```suggestion
		return fmt.Errorf("making Read request on Azure Firewall %s : %+v", *id, err)
```

`%q` here will output this wrapped in quotes - `%s` will output the value verbatim which is what we what here (the pointer de-reference here ensures a value is output rather than the points too fwiw